### DISCUSSION TITLE
[@types/oracledb] missing property fetchTypeHandler in type ExecuteOptions

### DISCUSSION THREAD

**Discussion by doberkofler** [DefinitelyTyped/DefinitelyTyped#68759] [Issues with a @types package]
The type definition of ExecuteOptions does not contain the property fetchTypeHandler as specified in the documentation.
https://node-oracledb.readthedocs.io/en/latest/user_guide/sql_execution.html

[a comment by an automated account is not shown]

**Comment by dannyb648** [comment]
Hi @doberkofler - this will be in 6.3.0 which I'm working on. Been delayed dealing with 6.2.1s fallout

**Comment by ritschwumm** [comment]
looks like it's there now in 6.3.0

**Reply by dannyb648** [reply]
Thanks for your input - hopefully 6.3.0 brings in a bunch of the requested fixes + some others we found

**Comment by ritschwumm** [comment]
interestingly, `ExecuteOptions.fetchTypeHandler` does not seem to support the type `(metaData:any) => void` as the global setting does any more. at runtime it apparently does, though (after convincing typescript with a cast to `any`).

**Comment by doberkofler** [comment]
@dannyb648 As @ritschwumm already noted, version 6.3.0 still cannot be used as intended [Using Fetch Type Handlers](https://node-oracledb.readthedocs.io/en/latest/user_guide/sql_execution.html#using-fetch-type-handlers).

The following example is valid and can be run but generates a type error:

```typescript
const result = await connection.execute(
    `SELECT * from myTable`,
    [],
    {
		resultSet: true,
		outFormat: oracledb.OUT_FORMAT_OBJECT,
		// @ts-expect-error
		// eslint-disable-next-line @typescript-eslint/consistent-return
		fetchTypeHandler: (metaData) => {
			if (metaData.dbType === oracledb.DB_TYPE_CLOB) {
				return {type: oracledb.STRING};
			}
		},
    }
);
```

**Reply by doberkofler** [reply]
@dannyb648 I provided the PR #69245 to fix the issue with `fetchTypeHandler`

**Reply by dannyb648** [reply]
Thanks for this, sorry for the delay, weve had some issues and only just getting on top of them. HAppy for this to go in and thanks for your help
